rfc6962-bis.xml

```diff
@@ -510,186 +596,170 @@ Section 8.2 of <xref target="RFC3447"/>) using a key of at least 2048 bits.
           </preamble>
           <artwork>
     enum {
-        v1(0), v2(1), (255)
-    } LeafVersion;
+        reserved(65535)
+    } SthExtensionType;
 
     struct {
-        uint64 timestamp;
-        LogEntryType entry_type;
-        select(entry_type) {
-            case x509_entry: CertInfo;
-            case precert_entry_V2: CertInfo;
-        } signed_entry;
-        SctExtensions extensions;
-    } TimestampedEntry;
+        SthExtensionType sth_extension_type;
+        opaque sth_extension_data&lt;0..2^16-1&gt;;
+    } SthExtension;
 
     struct {
-        LeafVersion version;
-        TimestampedEntry timestamped_entry;
-    } MerkleTreeLeaf;</artwork>
+        LogID log_id;
+        uint64 timestamp;
+        uint64 tree_size;
+        NodeHash root_hash;
+        SthExtension sth_extensions&lt;0..2^16-1&gt;;
+        digitally-signed struct {
+            TransItem merkle_tree_head;
+        } signature;
+    } SignedTreeHeadDataV2;</artwork>
         </figure>
         <t>
-          Here, <spanx style="verb">version</spanx> is the version of the MerkleTreeLeaf structure. This version is v2. Note that <xref target="RFC6962">MerkleTreeLeaf v1</xref> had another layer of indirection which is removed in v2.
+          <spanx style="verb">log_id</spanx> is this log's unique ID, encoded in an opaque vector as described in <xref target="log_id"/>.
         </t>
         <t>
-          <spanx style="verb">timestamp</spanx> is the timestamp of the corresponding SCT issued for this certificate.
+          <spanx style="verb">timestamp</spanx> matches the timestamp from the <spanx style="verb">TreeHeadDataV2</spanx> structure encapsulated in <spanx style="verb">merkle_tree_head</spanx>. This timestamp MUST be at least as recent as the most recent SCT timestamp in the tree. Each subsequent timestamp MUST be more recent than the timestamp of the previous update.
         </t>
         <t>
-          <spanx style="verb">entry_type</spanx> is the type of entry stored in <spanx style="verb">signed_entry</spanx>. New <spanx style="verb">LogEntryType</spanx> values may be added to <spanx style="verb">signed_entry</spanx> without increasing the <spanx style="verb">MerkleTreeLeaf</spanx> version. <xref target="client_messages"/> explains how clients should handle unknown entry types.
+          <spanx style="verb">tree_size</spanx> matches the tree size from the <spanx style="verb">TreeHeadDataV2</spanx> structure encapsulated in <spanx style="verb">merkle_tree_head</spanx>.
         </t>
         <t>
-          <spanx style="verb">signed_entry</spanx> is the <spanx style="verb">signed_entry</spanx> of the corresponding SCT.
+          <spanx style="verb">root_hash</spanx> matches the root hash from the <spanx style="verb">TreeHeadDataV2</spanx> structure encapsulated in <spanx style="verb">merkle_tree_head</spanx>.
```

Comment:
Addressed in https://github.com/robstradling/certificate-transparency-rfcs/commit/c42b1ea0f759c380708e5d0886dc726c127141a8